crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -139,72 +142,70 @@ pub fn update_frusta<T: Component + CameraProjection + Send + Sync + 'static>(
-        {
-            if !visibility.is_visible {
-                continue;
-            }
-
-            let entity_mask = maybe_entity_mask.copied().unwrap_or_default();
-            if !view_mask.intersects(&entity_mask) {
-                continue;
-            }
-
-            // If we have an aabb and transform, do frustum culling
-            if let (Some(model_aabb), None, Some(transform)) =
-                (maybe_aabb, maybe_no_frustum_culling, maybe_transform)
-            {
-                let model = transform.compute_matrix();
-                let model_sphere = Sphere {
-                    center: model.transform_point3a(model_aabb.center),
-                    radius: (Vec3A::from(transform.scale) * model_aabb.half_extents).length(),
-                };
-                // Do quick sphere-based frustum culling
-                if !frustum.intersects_sphere(&model_sphere, false) {
-                    continue;
+        let viz = Arc::new(RwLock::new(Vec::new()));
+
+        visible_entity_query.par_for_each_mut(
+            &pool,
+            32,
+            move |(
+                entity,
+                visibility,
+                mut computed_visibility,
+                maybe_entity_mask,
+                maybe_aabb,
+                maybe_no_frustum_culling,
+                maybe_transform,
+            )| {
+                computed_visibility.is_visible = false;
+                if visibility.is_visible {
+                    let entity_mask = maybe_entity_mask.copied().unwrap_or_default();
+                    if view_mask.intersects(&entity_mask) {
+                        let mut skip = false;
+                        // If we have an aabb and transform, do frustum culling
+                        if let (Some(model_aabb), None, Some(transform)) =
+                            (maybe_aabb, maybe_no_frustum_culling, maybe_transform)
+                        {
+                            let model = transform.compute_matrix();
+                            let model_sphere = Sphere {
+                                center: model.transform_point3a(model_aabb.center),
+                                radius: (Vec3A::from(transform.scale) * model_aabb.half_extents)
+                                    .length(),
+                            };
+                            // Do quick sphere-based frustum culling
+                            if !frustum.intersects_sphere(&model_sphere, false) {
+                                skip = true;
+                            }
+                            // If we have an aabb, do aabb-based frustum culling
+                            if !frustum.intersects_obb(model_aabb, &model, false) {
+                                skip = true;
+                            }
+                        }
+
+                        if !skip {
+                            computed_visibility.is_visible = true;
+                            viz.write().push(entity);
```

Comment:
I was expecting `viz` to be written into the visible entities at some point but it doesn’t look like anything is done with it?

Reply:
Looks like you were looking at an older commit. This was handled in 2a50ce81edc9706830910d1da12bbe27732d1e39